js/compiler.ts

```diff
@@ -58,6 +110,70 @@ interface ModuleMetaData {
 }
 
 class Host implements ts.CompilerHost {
+  private readonly _options: ts.CompilerOptions = {
+    allowJs: true,
+    allowNonTsExtensions: true,
+    checkJs: true,
+    esModuleInterop: true,
+    module: ts.ModuleKind.ESNext,
+    outDir: OUT_DIR,
+    resolveJsonModule: true,
+    sourceMap: true,
+    stripComments: true,
+    target: ts.ScriptTarget.ESNext
+  };
+
+  /** Take a configuration string, parse it, and use it to merge with the
+   * compiler's configuration options.  The method returns an array of compiler
+   * options which were ignored, or `undefined`.
+   */
+  configure(path: string, configurationText: string): string[] | undefined {
+    // this._log("compile.configure", path);
+    const { config, error } = ts.parseConfigFileTextToJson(
+      path,
+      configurationText
+    );
+    if (error) {
+      this._logDiagnostics([error]);
+    }
+    const { options, errors } = ts.convertCompilerOptionsFromJson(
+      config.compilerOptions,
+      cwd()
+    );
+    if (errors.length) {
+      this._logDiagnostics(errors);
+    }
+    const ignoredOptions: string[] = [];
+    for (const key of Object.keys(options)) {
```

Comment:
Use `for in` instead?

Reply:
Why?

Reply:
it's more `ES` way but it works too.
like:
`for (const key of Object.keys(options)) {` == `for (const key in options) {`